geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterQueryPlanner.scala

```diff
@@ -0,0 +1,159 @@
+    case shorten if expectedLen < hash.length  => new ARange(new Text(s"$res~${hash.substring(0, expectedLen)}"))
+    case lengthen if expectedLen > hash.length => new ARange(new Text(s"$res~$hash"), new Text(s"$res~$hash~"))
+  }
+
+  def getQueryPlan(rq: RasterQuery, resAndGeoHashMap: ImmutableSetMultimap[Double, Int]): Option[QueryPlan] = {
+    val availableResolutions = resAndGeoHashMap.keySet().toList.sorted
+
+    // Step 1. Pick resolution
+
+    val selectedRes: Double = selectResolution(rq.resolution, availableResolutions)
+    val res = lexiEncodeDoubleToString(selectedRes)
+
+    // Step 2. Pick GeoHashLength
+    val GeoHashLenList = resAndGeoHashMap.get(selectedRes).toList
+    val expectedGeoHashLen = if (GeoHashLenList.isEmpty) {
+      0
+    } else {
+      GeoHashLenList.max
+    }
+
+    // Step 3. Given an expected Length and the query, pad up or down the CAGH
+    val closestAcceptableGeoHash = GeohashUtils.getClosestAcceptableGeoHash(rq.bbox)
+
+    val hashes: List[String] = closestAcceptableGeoHash match {
+      case Some(gh) =>
+        val preliminaryHashes = List(gh.hash)
+        if (rq.bbox.equals(gh.bbox) || gh.bbox.covers(rq.bbox)) {
+          preliminaryHashes
+        } else {
+          val touching = TouchingGeoHashes.touching(gh).map(_.hash)
+          (preliminaryHashes ++ touching).distinct
+        }
+      case _ => Try {BoundingBox.getGeoHashesFromBoundingBox(rq.bbox) } getOrElse List.empty[String]
+    }
+
+    logger.debug(s"RasterQueryPlanner: BBox: ${rq.bbox} has geohashes: $hashes, and has encoded Resolution: $res")
+    logger.debug(s"Scanning at res: $selectedRes, with hashes: $hashes")
+    val r = hashes.map { gh => modifyHashRange(gh, expectedGeoHashLen, res) }.distinct
+
+    if (r.isEmpty) {
+      logger.debug(s"RasterQueryPlanner: Query was invalid given BBox: ${rq.bbox}")
+      None
+    } else {
+      // of the Ranges enumerated, get the merge of the overlapping Ranges
+      val rows = ARange.mergeOverlapping(r)
+      logger.debug(s"Scanning with ranges: $rows")
+      // setup the RasterFilteringIterator
+      val cfg = new IteratorSetting(90, "raster-filtering-iterator", classOf[RasterFilteringIterator])
+      configureRasterFilter(cfg, AccumuloRasterQueryPlanner.constructRasterFilter(rq.bbox.geom, rasterSft))
+      configureRasterMetadataFeatureType(cfg, rasterSft)
+
+      // TODO: WCS: setup a CFPlanner to match against a list of strings
+      // ticket is GEOMESA-559
+      Some(BatchScanPlan(null, rows, Seq(cfg), Seq.empty[Text], null, -1, hasDuplicates = false))
+    }
+  }
+
+  def selectResolution(suggestedResolution: Double, availableResolutions: List[Double]): Double = {
+    logger.debug(s"RasterQueryPlanner: trying to get resolution $suggestedResolution " +
+      s"from available Resolutions: ${availableResolutions.sorted}")
+    val ret = availableResolutions match {
+      case empty if availableResolutions.isEmpty   => 1.0
```

Comment:
i think the consensus we reached at one point was prefer if-else to pattern matching if you are not unapplying anything - which is the case here